src/components/panel/panel.js

```diff
@@ -975,12 +986,16 @@ MdPanelRef.prototype.hide = function() {
   var self = this;
   return this._$q(function(resolve, reject) {
     var done = self._done(resolve, self);
+    var onRemoving = self._config['onRemoving'] || angular.noop;
+
 
     self._$q.all([
       self._backdropRef ? self._backdropRef.hide() : self,
       self._animateClose().then(function() { self.addClass(MD_PANEL_HIDDEN); },
           reject)
-    ]).then(done, reject);
+    ]).then(onRemoving)
```

Comment:
This line should be moved up to: 
self._animateClose()
    .then(onRemoving)
    .then(function() { self.addClass(MD_PANEL_HIDDEN); })
    .catch(reject)